### PR TITLE
4489 Enhance hyperlinks of config syntax doc

### LINKED ISSUE #4489: Hyperlink in the config syntax doesn't work
**Describe the bug**
Thanks for Can's feedback, all the hyperlinks within "Syntax examples explained" in the config syntax page don't work:
https://docs.monai.io/en/latest/config_syntax.html#1--to-interpolate-with-python-objects

It may because the index or special character in the title names.
Would be nice to update them.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #4489 .

### Description
This PR fixed the hyperlinks of config syntax doc.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: docs/source/config_syntax.md [hunk 1]
```diff
@@ -12,10 +12,10 @@ Content:
 
 - [A basic example](#a-basic-example)
 - [Syntax examples explained](#syntax-examples-explained)
-  - [`@` to interpolate with Python objects](#1--to-interpolate-with-python-objects)
```
Comment (ericspod) [inline docs/source/config_syntax.md:15]: We had discussed the wording here earlier, on further thought perhaps this should be "@ to reference Python objects in configurations" and then using the word reference/referencing elsewhere to describe what @ is doing.

File: docs/source/config_syntax.md [hunk 2]
```diff
@@ -164,6 +164,7 @@ python -m monai.bundle COMMANDS
 where `COMMANDS` is one of the following: `run`, `verify_metadata`, `ckpt_export`, ...
 (please see `python -m monai.bundle --help` for a list of available options).
 
+The commands support many fexible use cases, like: override config at runtime, predefine CLI args in a file, etc.
```
Comment (wyli) [inline docs/source/config_syntax.md:167]:
```suggestion
The CLI supports flexible use cases, such as overriding configs at runtime and predefining arguments in a file.
```

File: docs/source/config_syntax.md [hunk 3]
```diff
@@ -182,3 +183,5 @@ Details on the CLI argument parsing is provided in the
   simple structures with sparse uses of expressions or references are preferred.
 - For `$import <module>` in the configuration, please make sure there are instructions for the users to install
   the `<module>` if it is not a (optional) dependency of MONAI.
+- As "#" and "$" can be syntax for `shell` or `CLI` tools, may need to add escape character or quotes for them
```
Comment (wyli) [inline docs/source/config_syntax.md:186]:
```suggestion
- As "#" and "$" might be interpreted differently by the `shell` or `CLI` tools, may need to add escape characters or quotes for them
```

File: docs/source/config_syntax.md (hunk 1 above)
Comment (wyli) [reply docs/source/config_syntax.md:15]: sure, sounds good to me.

File: docs/source/config_syntax.md (hunk 2 above)
Comment (Nic-Ma) [reply docs/source/config_syntax.md]: updated in the latest commit.

Thanks.

File: docs/source/config_syntax.md (hunk 1 above)
Comment (Nic-Ma) [reply docs/source/config_syntax.md:15]: Hi @ericspod @wyli ,

Updated in the latest commit.
Could you please help review it again?

Thanks in advance.

**Review by wyli — APPROVED** [review]

**Pull request merged**